File: Andromeda.AvaloniaApp/Helper/DownloadStatus.cs
using static GogApi.DotNet.FSharp.GalaxyApi;

namespace Andromeda.AvaloniaApp.Helpers
{
    public class DownloadStatus
    {
        public string GameTitle { get; private set; }
        public string FilePath { get; private set; }
        public float FileSize { get; private set; }
        public float Downloaded { get; private set; }
        public bool Installing { get; private set; }

        public DownloadStatus(string gameTitle, string path, float fileSize)
        {
            this.GameTitle = gameTitle;
            this.FilePath = path;
            this.FileSize = fileSize;
            this.Downloaded = 0;
            this.Installing = false;
        }

        public void UpdateDownloaded(float downloaded)
        {
            this.Downloaded = downloaded;
        }

        public void IndicateInstalling()
        {
            this.Installing = true;
        }
    }
}

File: Andromeda.AvaloniaApp.FSharp/Helpers/DownloadStatus.fs
namespace Andromeda.AvaloniaApp.FSharp.Helpers

open ReactiveUI

type DownloadStatus(gameTitle, path, fileSize) =
    inherit ReactiveObject()

    let mutable downloaded = 0
    let mutable installing = false

    member val GameTitle: string = gameTitle
    member val FilePath: string = path with get,set
    member val FileSize: float = fileSize
    member this.Downloaded
        with get () = downloaded
        and private set value = this.RaiseAndSetIfChanged(&downloaded, value) |> ignore
    member this.Installing
        with get () = installing
        and private set value = this.RaiseAndSetIfChanged(&installing, value) |> ignore

    member this.UpdateDownloaded downloaded = this.Downloaded <- downloaded
    member this.IndicateInstalling () = this.Installing <- true
